src/components/facets/FacetRow.tsx

```diff
@@ -8,9 +9,15 @@ export interface IFacetRowProps extends React.ClassAttributes<FacetRow> {
   isChecked: boolean;
 }
 
+export const MAX_NAME_LENGTH: number = 25;
+
 export class FacetRow extends React.Component<IFacetRowProps, any> {
 
   render() {
+    const label: JSX.Element = this.props.facetRow.formattedName.length > MAX_NAME_LENGTH
+      ? <Tooltip title={this.props.facetRow.formattedName} placement='top'>{this.props.facetRow.formattedName}</Tooltip>
+      : <span>{this.props.facetRow.formattedName}</span>;
```

Comment:
This could be done in css instead and would be more precise because 25 is a length that is more of an average than one that will work in every case. It depends mostly on the width of the facet.